src/containers/booksList.js

```diff
@@ -2,36 +2,65 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import { removeBook } from '../actions/index';
+import { removeBook, changeFilter } from '../actions/index';
 import Book from '../components/book';
+import CategoryFilter from '../components/CategoryFilter';
 
-const BooksList = ({ books, removeBook }) => {
-  const rows = [];
-  books.map(book => rows.push(
-    <Book
-      key={book.id}
-      book={book}
-      removeBook={removeBook}
-    />,
-  ));
-
-  return (
-    <div className="BooksList">
-      <table>
-        <thead>
-          <tr>
-            <th>Book ID</th>
-            <th>Title</th>
-            <th>Category</th>
-          </tr>
-        </thead>
-        <tbody>
-          {rows}
-        </tbody>
-      </table>
-    </div>
-  );
-};
+function filterdBooks(books, filter) {
+  return filter === 'All' ? books : books.filter(book => book.category === filter);
+}
+
+class BooksList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      filter: 'All',
+    };
+    this.filterChanged = this.filterChanged.bind(this);
+  }
+
+  filterChanged(event) {
+    this.setState({
+      filter: event.target.value,
+    });
+  }
+
+  render() {
+    const rows = [];
+    const { books } = this.props;
+    const { filter } = this.state;
+    const filteredBooks = filterdBooks(books, filter);
+    const { removeBook } = this.props;
```

Comment:
### [Optional]
let's practice [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) code.
we could join the props.
`const { books, removeBook } = ....`